app/services/candidate_interface/continuous_applications/submit_application_choice.rb

```diff
@@ -0,0 +1,32 @@
+module CandidateInterface
+  module ContinuousApplications
+    class SubmitApplicationChoice
+      attr_reader :application_choice, :application_form, :inactive_date_calculator
+      delegate :inactive_date, :inactive_days, to: :inactive_date_calculator
+
+      def initialize(application_choice, inactive_date_calculator: InactiveDateCalculator)
+        @application_choice = application_choice
+        @application_form = application_choice.application_form
+        @inactive_date_calculator = inactive_date_calculator.new(application_choice:, effective_date:)
+      end
+
+      def call
+        raise ApplicationNotReadyToSendError, application_choice unless application_choice.unsubmitted?
+
+        ActiveRecord::Base.transaction do
```

Comment:
We reverted the transaction block on the current submit service, due to it impacting the mailers. Wdyt tomas?

Reply:
I think the transaction approach we should follow to make sure we don't have candidate in weird states if something goes wrong in the submission.
The reason now that I am not creating classes that runs updates is that we can have control in this SubmitApplicationChoice class where we can pass directyl to the mailer the values that we would be updating in the DB.

So before in the mailers:

```
application_choice.reject_by_default_at => nil because the transaction didn't commit so many bugs
```

In the future:

```
  # we have the inactive date that we will be sending to the DB so all works in transactional matter
  CandidateMailer.application_submitted(application_form, inactive_date:).deliver_later
```

With this we are safe that if something goes wrong everything will rollback and in the code we don't use any columns from the DB **before** the transaction is commited to the DB.